### PR TITLE
Add Stylelint CSS linter

### LINKED ISSUE #266: Add a CSS Linter
To catch issues like:

> I guess this file is not preprocessed since it's in the `static` folder? Then you should use `/* … */` for comments, CSS doesn't support `//`.

_Originally posted by @FloEdelmann in https://github.com/OpenLightingProject/rdm-app/pull/265#discussion_r361483328_

### PR DESCRIPTION
**Pull request by FloEdelmann**
Closes #266. References https://github.com/OpenLightingProject/open-fixture-library/pull/1163.

### CODE REVIEW

File: Gruntfile.js [hunk 1]
```diff
@@ -2,6 +2,8 @@ module.exports = function(grunt) {
   'use strict';
   require('google-closure-compiler').grunt(grunt);
 
+  grunt.loadNpmTasks('grunt-stylelint');
```
Comment (peternewman) [inline Gruntfile.js]: Could you put this with the other load tasks, or move them all up here if you feel they should be above the configs.

File: static/css/main.css [hunk 2]
```diff
@@ -89,18 +85,6 @@ a:hover .missing_image {
     border-bottom: 1px solid #e4e9fe;
 }
 
-.sorter_icon {
```
Comment (peternewman) [inline static/css/main.css]: I assume we were just duplicating this from elsewhere?

**Comment by FloEdelmann** [comment]
Double-slash comments are detected:

Screenshot_2020-01-14_11-11-14

File: Gruntfile.js (hunk 1 above)
Comment (FloEdelmann) [reply Gruntfile.js]: I've moved it below to match the [sample Gruntfile](https://gruntjs.com/sample-gruntfile) from the Grunt website.

File: static/css/main.css (hunk 2 above)
Comment (FloEdelmann) [reply static/css/main.css]: Yes, it was exactly duplicated from Bootstrap's `.glyphicon` ruleset.

Screenshot_2020-01-14_11-32-17

**Comment by peternewman** [comment]
Do you fancy doing OLA too @FloEdelmann ? :laughing:

**Comment by FloEdelmann** [comment]
Sure, though I need a little more guidance on where to put the stylelint call and which files to lint.

From the [list of CSS files](https://github.com/OpenLightingProject/ola/search?l=css), it looks like only [javascript/new-src/css/style.css](https://github.com/OpenLightingProject/ola/blob/master/javascript/new-src/css/style.css) is self-authored, the other ones are copied from other frameworks, right? So I guess linting all CSS files in [javascript/](https://github.com/OpenLightingProject/ola/blob/master/javascript/) should be the right way to go, to also catch future files? Or are built files also put there?

**Comment by peternewman** [comment]
> Sure, though I need a little more guidance on where to put the stylelint call and which files to lint.

Great!

> From the [list of CSS files](https://github.com/OpenLightingProject/ola/search?l=css), it looks like only [javascript/new-src/css/style.css](https://github.com/OpenLightingProject/ola/blob/master/javascript/new-src/css/style.css) is self-authored, the other ones are copied from other frameworks, right?

Yeah that looks about right, I thought there was some styling for the old UI and the RDM test tools but it does indeed look like they're just stock UI config. Actually it's in the HTML, can we lint that, I guess not easily?:
https://github.com/OpenLightingProject/ola/blob/master/olad/www/ola.html

> So I guess linting all CSS files in [javascript/](https://github.com/OpenLightingProject/ola/blob/master/javascript/) should be the right way to go, to also catch future files? Or are built files also put there?

Sounds sensible, although actually it might be worth just doing javascript/new-src especially as I assume you'll add it to the Gruntfile in there, which already has a list of CSS files to process:
https://github.com/OpenLightingProject/ola/blob/master/javascript/new-src/Gruntfile.js

**Pull request merged**